### PR TITLE
Expanded state supervisables - current / total / new

### PR DESCRIPTION
**Pull request by meirhalachmi**
This PR includes Freiman's dump-to-CSV PR

### CODE REVIEW

File: src/corona_hakab_model/consts.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class Consts(NamedTuple):
 
     # attributes and default values:
 
-    total_steps: int = 350
+    total_steps: int = 40
```
Comment (tal-nc) [inline src/corona_hakab_model/consts.py:32]: switch this back to a higher number before merging

File: src/corona_hakab_model/agent.py [hunk 2]
```diff
@@ -59,21 +59,27 @@ def get_infection_ratio(self):
 
 
 class Circle:
-    __slots__ = "kind", "agent_count"
+    __slots__ = "kind", "agent_count", "added_total", "removed_total"
```
Comment (tal-nc) [inline src/corona_hakab_model/agent.py:62]: tracking circle has ever_visited, which is the same as added total. removed total is just ever_visited - agent_count

File: src/corona_hakab_model/consts.py (hunk 1 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/consts.py:32]: oops :laughing:

File: src/corona_hakab_model/agent.py (hunk 2 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/agent.py:62]: Missed `ever_visited`, and you're right about the `removed_total`. I'll change it

File: src/corona_hakab_model/supervisor.py [hunk 3]
```diff
@@ -34,129 +34,41 @@ def snapshot(self, manager):
...
+            published_data = [v.publish() for v in self.supervisables]
+            csv_writer = csv.writer(output_file)
+            # write data rows
+            for i in range(self.manager.consts.total_steps + 1):
+                row = [i]
+                if i == 0:  # header row
+                    row = ["days"]
+                for data in published_data:
+                    # first column is always the #day
+                    row = row + data[i][1:]
```
Comment (mishana) [inline src/corona_hakab_model/supervisor.py:51]: isn't this gonna attach real data to the header row as well?

Comment (tal-nc) [reply src/corona_hakab_model/supervisor.py:51]: i think that the first row of each supervisable is its name, which should be in the header row

Comment (meirhalachmi) [reply src/corona_hakab_model/supervisor.py:51]: Anyway I'm switching most of this code to use Pandas

**Pull request merged**